src/main/java/org/opensearch/ad/feature/AbstractRetriever.java

```diff
@@ -30,8 +32,49 @@
 import org.opensearch.search.builder.SearchSourceBuilder;
 
 public abstract class AbstractRetriever {
-    protected double parseAggregation(Aggregation aggregation) {
+    protected double parseAggregation(Aggregation aggregationToParse) {
         Double result = null;
+        /* example InternalSingleBucketAggregation: filter aggregation like
+           "t_shirts": {
+                "filter": {
```

Comment:
Detector data filter can also filter data, why we need to support filter in `aggregation_query`?

Reply:
We need this when we have multiple features with different filters.